Revise 0315 - Add RPC conflict management

Akihiro-Miyazaki opened the pull request:
## Introduction
This is an update of SDL-0315.

## Motivation
Modify to remove SDLManager from Figure 1.

## Proposed solution
Change the stored file of the configuration data from a new file to policy table.
Change the word from "RPC Conflict Management Configuration File" to "RPC Conflict Management Configuration Table".
Modify the RPC Conflict Management Configuration Table according to [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/1068#issuecomment-704467722).
Also, modify "RPC Conflict Management Module" to change the word from "configuration file" to "configuration table".

## Potential downsides
Add the potential downsides according to [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/1068#issuecomment-704467722).
However, remove the sentence "This also adds new config files that are not updatable outside of a firmware update (i.e. it's not a policy change).".

## Impact on existing code
Add HMI.

Akihiro-Miyazaki commented:
Hi, @theresalech  -san.
This documentation was revised according to [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/1068#issuecomment-704467722).
Could you please review my proposal.
Thank you.

theresalech commented on proposals/0315-Add-RPC-Conflict-Management.md:
```diff
@@ -22,22 +22,30 @@ The example above shows that the latter RPC `PerformAudioPassThru` is displayed
+To implement the RPC conflict management, we will add a new RPC conflict management module, `InterruptManager` to SDL Core, and add `InterruptManagerConfig` to policy table.
 
-RPC priority table and `AppHMIType` priority table are specified in `InterruptManagerConfig`. RPC priority table and `AppHMIType` priority table are tables that each set the priority of RPC and app type (`appHMIType`). By modifying `InterruptManagerConfig`, an OEM can receive the expected request from SDL Core during RPC conflict. On the other hand, `InterruptManager` reads `InterruptManagerConfi ...
+RPC priority table and `App Priority` table are specified in `InterruptManagerConfig`. RPC priority table and `App Priority` table are tables that each set the priority of RPC and app. By modifying `InterruptManagerConfig`, an OEM can receive the expected request from SDL Core during RPC conflict. On the other hand, `InterruptManager` reads `InterruptManagerConfig` in policy table during the SDL  ...
```
Can you please advise which [agreed upon revision](https://github.com/smartdevicelink/sdl_evolution/issues/1068#issuecomment-704467722) requested changing `AppHMIType` priority table to `App Priority` table?

Akihiro-Miyazaki replied:
@theresalech -san,
The RPC priority of previous proposal is below.
1. RPC priority
2. appHMIType priority
However, In review, the RPC priority order was chaged as below.
1. Apps with app priority EMERGENCY
2. RPC priority
3. App priority(non-emergency)
4. HMI Status
Since the priority is replaced from appHMIType to app priority, the appHMIType was not used in the priority order. Therefore, I removed the appHMIType section.

theresalech reviewed (changes requested):
@Akihiro-Miyazaki I am still in the process of reviewing this PR but would like to get an answer to the question posed in my first comment before proceeding.

In addition to my comment left in-line, I noticed a few [agreed upon revisions](https://github.com/smartdevicelink/sdl_evolution/issues/1068#issuecomment-704467722) that are not reflected in this PR. Can you please make sure the following are reflected, and let me know when this PR is ready for re-review? If they are already reflected, please provide me with the location of the change within the PR. Thank you!
- (Item 4) Specify that the RPC priority table overrides the App HMI type table.
- (Item 5) Add the following sentence: "For apps with multiple AppHMITypes, whichever among the HMI types is the highest priority will be used for the entire app" to "`appHMIType` Priority Table" section
- (Item 10, part 2) Add the following to the "Potential Downsides" section:
> 2. This also adds new config files that are not updatable outside of a firmware update (i.e. it's not a policy change).

Akihiro-Miyazaki commented:
@theresalech -san,
(Item 4) is described in Line 36.
 "The RPC priority is overridden as the order below."
 "the App HMI type table" was removed from this proposal. Therefore, I modified to above sentence.

(Item 5) is removed from this proposal.
 Since this proposal is replaced from "appHMIType priority" to "app priority", there is no description of appHMIType. So I can not add this sentence.

(Item 12 part 2) is removed from this proposal.
 Since this proposal is changed from a new file to policy table file, I removed this comment from the proposal documentation.

theresalech commented on proposals/0315-Add-RPC-Conflict-Management.md:
```diff
@@ -3,7 +3,7 @@
+* Impacted Platforms: [ Core / HMI ]
 
 ## Introduction
 This proposal is to add a function to SDL Core that manages concurrent RPC conflicts according to priority. This will allow OEMs to receive the expected request from SDL Core during RPC conflict by modifying the configuration file in which the priority is set.
```
Reference to the "configuration file" should be removed, and Introduction should be further updated to reflect revised proposed solution.

Akihiro-Miyazaki replied:
@theresalech -san,
Thank you for your review again.
I will modify to change from "configuration file" to " configuration table in the policy table".
Does  this way matched your advise?
Also, I will modify to add the "/ Policy Server" onto Impacted Platforms.
Anyway, I will modify this proposal and make a commit it.
Could you please confirm it again?
Thank you.

theresalech replied:
I've provided a suggested change to the `Introduction` section.

theresalech commented on proposals/0315-Add-RPC-Conflict-Management.md:
```diff
@@ -114,79 +123,44 @@ OEMs can restore the deleted `UI.Slider` and modify its priority. For example, i
-<b>1.Loading of configuration file</b><br>
-`InterruptManagerConfig` is loaded during the startup of SDL Core.
+<b>1.Loading of configuration table</b><br>
+`InterruptManagerConfig` is loaded from policy table during the startup of SDL Core.
```
Should `InterruptManagerConfig` be `RpcPriority` here?

Akihiro-Miyazaki replied:
@theresalech -san,
Thank you for your response.
In this matter, I think OK as it is.
Because I described "RPC priority table and App Priority table are specified in InterruptManagerConfig." in this proposal.

theresalech commented on proposals/0315-Add-RPC-Conflict-Management.md:
```diff
@@ -22,22 +22,30 @@ The example above shows that the latter RPC `PerformAudioPassThru` is displayed
+- `AppPriority` table
+- HMI status table
+
+The RPC priority is overridden as the order below. Priority order will shift from 1 to 2, from 2 to 3 and from 3 to 4, if the RPCs are same priority.
```
```suggestion
The RPC priority will be determined by the order below. Priority order will shift from 1 to 2, from 2 to 3 and from 3 to 4, if the RPCs are same priority.
```

theresalech reviewed (changes requested):
@Akihiro-Miyazaki, thank you for your responses. I've verified no further action is required for Items 4, 5, and 10 part 2, and completed my review. Please let me know if you have any questions or concerns with addressing my feedback. Thanks!

Akihiro-Miyazaki commented:
Hi, @theresalech -san,
Thank you for your cooperation as always.
I left my comments and updated the proposal.
Please confirm them. 
Thank you.

theresalech reviewed (changes requested):
Hi @Akihiro-Miyazaki, thank you! I've left one more suggested change based on your latest commits. Please let me know if you have any questions or concerns with addressing. Thanks!

Akihiro-Miyazaki commented:
Hi, @theresalech -san,
Thank you for your suggestion as always.
I have no question about your suggestion.
Therefore, I made a commit it.
Thank you.
